oeps/oep-0045/decisions/0001-tutor-as-replacement-for-edx-configuration.rst

```diff
@@ -0,0 +1,328 @@
+can be disabled locally and easily `configured to use an externally hosted service
+<https://docs.tutor.overhang.io/configuration.html#mysql>`__:
+
+-  MySQL
+-  Elasticsearch
+-  MongoDB
+-  Redis
+-  SMTP
+
+
+Multi-server
+````````````
+
+Single-server deployments are less than ideal in terms of high availability and scalability, so for larger deployments
+something different is needed. And given that Tutor deployments use Docker images, Kubernetes would be the first choice.
+(Not least of which because Kubernetes is also what edX `has started to use
+<https://discuss.openedx.org/t/lets-talk-about-the-native-installation/3269/14>`__.)
+
+Tutor does come with basic support for Kubernetes deployment.  ``tutor k8s quickstart`` on a local `MicroK8s
+<https://microk8s.io/>`__ 1.8 installation did not work out of the box, though. There is a race condition during the
+startup of the MySQL container that requires relaunching the command, and one has to modify the MinIO hostname manually
+to avoid a failure during the initial LMS job. There were further MinIO name resolutions issues that precluded a fully
+working environment: there was no time to ascertain whether the cause is ultimately with Tutor or with the tested
+microk8s setup.
+
+It is mentioned at the top of the `Tutor Kubernetes documentation <https://docs.tutor.overhang.io/k8s.html>`__ that "the
+local Open edX install was designed such that users with no prior experience with system administration could still
+launch an Open edX platform. It is not the case for the installation method outlined here." While the Tutor project has
+clearly expended effort in getting it to work with Kubernetes, if it is to fully replace ``edx/configuration`` as the
+go-to community installation for production use, it is suggested that the Kubernetes documentation be extended with more
+detailed information on how to run and maintain Open edX in such an environment, and that Kubernetes support be made a
+first-class citizen.
+
+Note that it is perfectly possible to bypass Tutor entirely when it comes to cluster management. One could use its image
+creation engine to build the requisite Docker images, then manage the Kubernetes cluster with something like Helm: there
+are reports of `this exact thing being done
+<https://discuss.openedx.org/t/lets-talk-about-the-native-installation/3269/2>`__).
+
+
+Criterion 2. Automated deployment and testing
+.............................................
+
+Launching a preconfigured instance from scratch is simple, as Tutor can either `load configuration
+<https://docs.tutor.overhang.io/configuration.html#configuration>`__ from environment variables, from locally stored
+YAML files, or via command line parameters. Automating a launch is therefore just as simple as launching an
+``edx/configuration`` Ansible run.
+
+Nevertheless, it is not currently possible to run edx-platform tests in a Tutor-created container, so automation for
+this particular purpose will require work.
```

Comment:
I did some preliminary testing with the `openedx-dev` Docker image and the results were surprisingly good:

```
$ tutor dev run lms bash
$ unset DJANGO_SETTINGS_MODULE
$ pytest openedx/core/djangoapps/waffle_utils/
...
============== 22 passed, 30 warnings in 6.68s ==============
```

Some other tests fail because of an incorrect MongoDb host setting. This is just an example of a test setting that needs to be correctly defined, but I suspect that getting unit tests to run will not require much more work.

Reply:
This is great news!  I didn't go very deep into this rabbit hole yet, but I'm glad to see it might not be all that hard.